ocs_ci/framework/pytest_customization/marks.py

```diff
@@ -591,3 +591,13 @@
 
 # Marks to identify the cluster type in which the test case should run
 runs_on_provider = pytest.mark.runs_on_provider
+
+baremetal_or_vsphere_deployment_required = pytest.mark.skipif(
+    config.ENV_DATA["platform"].lower() not in ["baremetal", "vsphere"],
+    reason="Test required baremetal or vsphere deployment.",
+)
+
+ui_deployment_required = pytest.mark.skipif(
+    not config.DEPLOYMENT["ui_deployment"],
```

Comment:
this ui_deployment flag is mainly considered for deployment part, in the jenkins job we propagate it to all relevant stages - so test_stage should probably have it as well.  The question is if we should really rely on this when this config value is mainly used for deployment - considering it's part of DEPLOYMENT section ;)  Which means if someone will run it outside our jenkins - and if they do not propagate this value the test will be skipped. Even they run locally after they deployed via UI manually and they forgot to pass this value as it's mainly meant for deployment.

Reply:
Hi @petr-balogh . I think who ever uses the test should know how to propagate the right values as our tests are full of marks. If someone doesn't know he should learn. beside that what are our options?

Reply:
We have discussed offline with Parag and he is trying to get from UI dev team info if there is a way to dynamically find out if cluster was deployed via UI. If there is such a way to figure it out, then this bill be the best way to check and skip this test based on it.

Even though I do the deployment via UI, let's say manually, I can still decide to uncheck the che
ckbox to not set RBD as default SC. Then this test is also not relevant I think. Current, behavior is:
<img width="1185" alt="Screenshot 2024-03-21 at 10 54 30" src="https://github.com/red-hat-storage/ocs-ci/assets/16169617/70793b5c-3ab7-4d12-8815-3d11c3ebb4ce">

That the checkbox is selected, and we don't do anything, then it will be used. Currently we don't have an automation to change it.

So this one is a bit tricky and what I am trying to say is that the config.DEPLOYMENT["ui_deployment"] value is mainly meant to be used for UI deployment test which does the deployment itself and can be sure that all values are correct for the deployment part. Hence discussed with @OdedViner to add some check for this feature directly to deployment itself, when this value can be used for sure, and it's a simple check to test this value is checked by default to True.  And we can at the end of deployment do additional check to really check the annotation of the SC is really set to be default.